geode-for-redis/src/main/java/org/apache/geode/redis/internal/data/RedisList.java

```diff
@@ -135,6 +161,27 @@ public int getDSFID() {
     return REDIS_LIST_ID;
   }
 
+  public synchronized int elementInsert(byte[] elementToInsert, byte[] referenceElement,
+      boolean before) {
+    for (int i = 0; i < elementList.size(); i++) {
+      if (Arrays.equals(elementList.get(i), referenceElement)) {
+        if (before) {
+          elementList.add(i, elementToInsert);
+          return i;
+        } else {
+          elementList.add(i + 1, elementToInsert);
+          return i + 1;
+        }
+      }
+    }
+
+    return -1;
+  }
+
+  public synchronized void elementInsert(byte[] toInsert, int index) {
```

Comment:
I could be wrong but this seems like overkill. Just have the single 'int elementInsert' method. Callers are free to ignore its result. If you do keep this you don't need it to be synchronized since it does not do anything that is not thread safe.

Reply:
this elementInsert is used when applying the delta. at that point the only information that we have is the index and the new value